src/blobs.rs

```diff
@@ -53,13 +52,14 @@ impl Iterator for OsmObjs {
 }
 
 /// Transforms a `Result<blob>` into a `Iterator<Item = Result<OsmObj>>`.
-pub fn result_blob_into_iter(result: Result<Blob>) -> OsmObjs {
+pub fn result_blob_into_iter(result: Result<Blob>) -> OsmObjsIter {
     match result.and_then(|b| ::reader::primitive_block_from_blob(&b)) {
-        Ok(block) => OsmObjs(OsmObjsImpl::OkIter(new_rent_osm_objs(block).map(Ok))),
-        Err(e) => OsmObjs(OsmObjsImpl::ErrIter(iter::once(Err(e)))),
+        Ok(block) => OsmObjsIter(OsmObjsImpl::OkIter(new_rent_osm_objs(block).map(Ok))),
+        Err(e) => OsmObjsIter(OsmObjsImpl::ErrIter(iter::once(Err(e)))),
     }
 }
 
-fn new_rent_osm_objs(block: PrimitiveBlock) -> rent::OsmObjs {
-    rent::OsmObjs::new(Box::new(block), blocks::iter)
+fn new_rent_osm_objs(block: PrimitiveBlock) -> OsmBlobObjs {
+    // blocks::iter
+    OsmBlobObjs::from_fn(block, |block| panic!())
```

Comment:
Panic?

Reply:
Oops, good catch! That's a leftover from the migration, will fix.